neo/Ledger/MemoryPool.cs

```diff
@@ -165,6 +171,31 @@ public IEnumerator<Transaction> GetEnumerator()
 
         IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
 
+        public BigInteger GetSenderFee(UInt160 sender)
+        {
+            _txRwLock.EnterReadLock();
+            bool recorded = _senderFee.TryGetValue(sender, out var value);
+            _txRwLock.ExitReadLock();
+            if (recorded)
+                return value;
+            else
+                return BigInteger.Zero;
+        }
+
+        private void AddSenderFee(Transaction tx)
+        {
+            if (_senderFee.TryGetValue(tx.Sender, out var value))
```

Comment:
No need for Read and write here?

Reply:
I think no, as the places the func called has been attached writelocks